wordpress-mobile/WordPress-Android
# Blogging reminders remove feature flag

This PR 
- Removes the `BloggingReminderFeatureConfig`, since it has been released over an year ago.
- Removes it from Site Settings on WP (See before image below, it was present under two categories)
- Updates Blogging Reminders to be JP exclusive

| WP Before | WP After |
|--------|--------|
| ![Screenshot_20230711_171025](https://github.com/wordpress-mobile/WordPress-Android/assets/990349/47b4eb51-2554-41b3-9125-f9944c194548)| ![Screenshot_20230711_165914](https://github.com/wordpress-mobile/WordPress-Android/assets/990349/089aa321-522e-4b40-8988-8e01c17be213)| 

To test:

The Blogging Reminders feature has two main UI components:

A one-time prompt shown per-blog to promote the feature.
A setting in Site Settings for the user to adjust when the local reminder notification is shown.

JP app
- Add a new site to the Jetpack app
- Publish a new Post (FAB -> Blog post)
- Verify that the Blogging Reminders prompt is shown
- Additionally, navigate to Menu -> Site Settings, and
- Verify Reminders setting is visible under **Blogging** category, and works

WP app
- Add a site to WP app
- Publish a Post (FAB -> Blog post)
- `Verify` that the Blogging Reminders prompt is `not shown`
- Additionally, navigate to Menu -> Site Settings, and
- `Verify` Reminders is `not shown` (as in after image above under neither General nor Blogging)

> **Note**
> This doesn't handle any self hosted related cases.  That'll be tackled in a separate PR

## Regression Notes
1. Potential unintended areas of impact
Tested on both WP and JP apps

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing, and unit tests

3. What automated tests I added (or what prevented me from doing so)
Updated unit tests

PR submission checklist:
